### ISSUE TITLE
Service name is not allowing underscore while consul accepts it

### ISSUE THREAD

**Issue by lchayoun** [hashicorp/nomad#6377]
If filing a bug please include the following:

### Nomad version
Nomad v0.9.5 (1cbb2b9a81b5715be2f201a4650293c9ae517b87)

### Operating system and Environment details
Ubuntu 16.04.6 LTS

### Issue
When submitting a job with service name that contains underscore, the following error is shown:

* Task group service validation failed: 1 error(s) occurred:

* Task Service_Name validation failed: 1 error(s) occurred:

* 1 error(s) occurred:

* service[0] "Service_Name" validation failed: 1 error(s) occurred:

* service name must be valid per RFC 1123 and can contain only alphanumeric characters or dashes: "Service_Name"

### Reproduction steps
Submit a job with service name that contains underscore

### Job file (if appropriate)
the example job name with service name `redis_cache` instead of `redis-cache`

### Nomad Client logs (if appropriate)
Not relevant

### Nomad Server logs (if appropriate)
Not relevant

**Comment by nickethier** [comment]
Hi @lchayoun 

The reason we enforce stricter validation here is to avoid potential issues with resolving the service name via DNS. See https://www.consul.io/docs/agent/services.html#service-and-tag-names-with-dns

Even without Consul's DNS interface, a common pattern is to use the service name as part of the host when hitting a load balancer like Fabio or Traefik. When the service name is not RFC 1123 compliant, these scenarios fail in odd and hard to debug ways.

In summary this is working as expected. If have a really compelling reason why Nomad shouldn't do this please let feel free to reopen this issue.
